alphagov/paas-admin
# [#173487093] Add support & signup forms

Massive thanks to @paroxp for his support and contribution to this PR.

What
----

This is an implementation of current support forms found on product pages
- https://www.cloud.service.gov.uk/support
- https://www.cloud.service.gov.uk/signup
- https://www.cloud.service.gov.uk/contact-us

into a public Paas admin /support route.

For `get/post` routes i opted to use different handler, so as to make it more readable for myself instead of having a single complex `/support/:route`  handler.

Content of forms, confirmation pages and emails follows what is currently there.

Why
----

We are building new product pages which are fully static HTML. Additionally, bringing support form into here should allow as to evolve these forms in the future with app and user data to make them even more effortless. 

How to review
-------------
- checkout `support-page` branch
- run `npm install`
- Run locally with `ZENDESK_API_TOKEN="{token}" ZENDESK_USERNAME="{user}" npm run start:with-stub-api` with creds from the password-store (or deploy to your env)
- check forms and error message
- evaluate code

Note: tried to split work into logical commits. later on changes applied across features so best to look at the combination of commit-by-commit and whole end result file

Who can review
---------------
not @kr8n3r 

Visuals
---------------

<details>
<summary>Visual changes</summary>
Header breakpoints have been adjusted to allow for more navigation elements<br/><br/>

![paas-header](https://user-images.githubusercontent.com/3758555/86099865-f4f59600-baaf-11ea-96e3-03e70c6c918a.gif)
</details>


<details>
<summary>Example form with error messages</summary>

![0 0 0 0_3000_support_something-wrong-with-service](https://user-images.githubusercontent.com/3758555/87635088-697a3700-c736-11ea-833c-57fd3e31c142.png)
</details>

<details>
<summary>Example confirmation page</summary>
Confirmation page content is different for each type of support request<br/><br/>

![0 0 0 0_3000_support_something-wrong-with-service (1)](https://user-images.githubusercontent.com/3758555/87635344-c970dd80-c736-11ea-98c4-29a8eb5608a1.png)
</details>

<details>
<summary>Zendesk emails</summary>
Example of Zendesk email content and subject line. <br/>

<img width="426" alt="Screenshot 2020-07-16 at 07 25 47" src="https://user-images.githubusercontent.com/3758555/87634837-f2dd3980-c735-11ea-9231-a9a6f7284148.png"> <br/>

<img width="362" alt="Screenshot 2020-07-16 at 07 26 56" src="https://user-images.githubusercontent.com/3758555/87634841-f40e6680-c735-11ea-8f81-b27e23a67389.png"><br/>

<img width="409" alt="Screenshot 2020-07-16 at 07 27 42" src="https://user-images.githubusercontent.com/3758555/87634843-f4a6fd00-c735-11ea-974d-8bad7b57aa5c.png"><br/>
</details>
